wazo_confd/plugins/line_extension/middleware.py

```diff
@@ -37,4 +37,8 @@ def delete_extension(self, line_id, extension_id, tenant_uuid, tenant_uuids):
         self.dissociate(
             line_id, extension_id, tenant_uuid, tenant_uuids, reset_autoprov=True
         )
-        extension_middleware.delete(extension_id, tenant_uuids)
+        try:
+            extension_middleware.delete(extension_id, tenant_uuids)
+        except ResourceError as e:
+            if not str(e).startswith('Resource Error - Extension is associated'):
+                raise e
```

Comment:
Instead of matching on exception message, how about creating an exception Subclass(`ExtensionAssociatedError`) and catching only that? I believe that would be less error prone, more concise.

Reply:
I totally agree implementing a dedicated exception would be less error prone. Unfortunately I cannot easily create a new Exception, as the ResourceError is created by the validator (wazo_confd/plugins/extension/validator.py). The validators are used when there is a creation, a deletion and an update through the DAOs. The validators are one of the pillars of the DAOs, so there is no easy way to replace/write a new Exception.

Reply:
From my understanding, LineExtensionAssociationValidator is only used by the LineExtensionService, so exceptions generated from the validator are only relevant to the service and layers above( API middleware/http resource definition) which use the service. I don't see the DAOs depending on ResourceError.
That being said, I understand not wanting to break the mold here.
But this looks like a bad "dependency" if the xivo_dao lib dictates how exceptions should be used by application-specific code that is not relevant to the DAOs.